vllm/transformers_utils/config.py

```diff
@@ -70,6 +71,27 @@ def get_config(model: str,
             logger.info("Updating %s from %r to %r", key,
                         getattr(config, key, None), value)
             config.update({key: value})
+
+    if trust_remote_code:
+        # With trust_remote_code, the config is typically an instance of a
+        # custom class imported from the HF modules cache.
+        #
+        # The class will not be importable in Ray workers by default (and won't
+        # exist at all on other nodes), which breaks serialization of the
+        # config. Here we tell the serialization library used by Ray to pass
+        # instances of these generated classes by value instead of by reference
+        # (eg. the class definition is serialized along with its data).
+        #
+        # See: https://github.com/cloudpipe/cloudpickle?tab=readme-ov-file#overriding-pickles-serialization-mechanism-for-importable-constructs
+        try:
+            import transformers_modules
+            ray.cloudpickle.register_pickle_by_value(transformers_modules)
```

Comment:
Can use `import cloudpickle` rather than `ray.cloudpickle`

Reply:
Ray vendors its own version of cloupickle:
https://github.com/ray-project/ray/tree/master/python/ray/cloudpickle

I've updated the code to register the by-value serialization for the ray vendored version and for the top-level version used by MP.